src/contractor.jl

```diff
@@ -3,13 +3,20 @@
 `Contractor` represents a `Contractor` from ``\\mathbb{R}^N`` to ``\\mathbb{R}^N``.
 Nout is the output dimension of the forward part.
 """
-struct Contractor{N, Nout, F1<:Function, F2<:Function, ex<:Union{Operation,Expr}}
+abstract type AbstractContractor end
+
+struct Contractor{N, Nout, F1<:Function, F2<:Function, ex<:Union{Operation,Expr}} <:AbstractContractor
     variables::Vector{Symbol}  # input variables
     forward::GeneratedFunction{F1}
     backward::GeneratedFunction{F2}
     expression::ex
 end
 
+struct BasicContractor{F1<:Function, F2<:Function} <:AbstractContractor
+    forward::GeneratedFunction{F1}
```

Comment:
We don't need the `code` that's stored in a `GeneratedFunction`; this should just be the function itself.

Reply:
The goal is to have an object that satisfies `isbitstype(C) == true`.